General coding practices/Test Automation (Mobile) - element IDs.md

```diff
@@ -0,0 +1,98 @@
+
+For list items that are not known in advance, the developer should implement an ID to the list container. After that, the testers should be able to get those elements as children of the container. Also, in case of adding IDs to individual list items that are not known in advance, it is alright if the same ID is implemented to them because the automation framework can locate such elements by combining the ID and text of the element.
+
+### Testing
+
+Implementation can be tested using [Accessibility Inspector](https://developer.apple.com/videos/play/wwdc2019/257/), and checking `Basic → Identifier`.
+
+<span style="display:block; border: 1px solid #e0e0e0; margin-top:15px; margin-bottom:15px; margin-left:auto; margin-right:auto; width:80%;">![iOS Accessibility Inspector](/img/test_automation/TA_iOS_Accessibility_Inspector.png)</span>
+
+## Android
+
+### XML architecture
+
+#### Static elements
+
+For static elements such as buttons, titles and others that are not dynamically added Android should use **resource ID**. This is already present on a lot of views and automated tests can make use of that.
+ 
+*Example:*
+
+`fragment_my_profile.xml` has the profile name of the currently logged-in user. Even though the text that is displayed can be different, the ID will always be the same for that text.
+
+<span style="display:block; border: 1px solid #e0e0e0; margin-top:15px; margin-bottom:15px; margin-left:auto; margin-right:auto; width:100%;">![Android Static Element Example](/img/test_automation/TA_Android_Static.png)</span>
+
+#### Dynamically added elements
+
+Dynamically added elements can be split into 2 categories. Ones that are known in advance and others that are unknown.
+
+##### Known in advance
+For the ones that are known in advance, they should be set in ids.xml file as item of type="id". After that, they can be programmatically set to the ID of an element that they represent.
+
+*Example:*
+
+`fragment_select_appliance_category.xml` contains `list_item_appliance_category.xml` as elements of a list. Since categories are known in advance they can be set up in `ids.xml` and added programmatically to the list item.
+
+<span style="display:block; border: 1px solid #e0e0e0; margin-top:15px; margin-bottom:15px; margin-left:auto; margin-right:auto; width:100%;">![Android Dynamic Element Example](/img/test_automation/TA_Android_Dynamic.png)</span>
+
+##### Unknown in advance
+
+For list items that are not known in advance, the developer should implement an ID to the list container. After that, the testers should be able to get those elements as children of the container. Also, in case of adding IDs to individual list items that are not known in advance, it is alright if the same ID is implemented to them because the automation framework can locate such elements by combining the ID and text of the element.
```

Comment:
```suggestion
For list items that are not known in advance, while developing ID should be added to the list container. After that, while testing we should be able to get those elements as children of the container. Also, in case of adding IDs to individual list items that are not known in advance, it is alright if the same ID is implemented to them because the automation framework can locate such elements by combining the ID and text of the element.
```

The same thing as above, what do you think?

Reply:
Updated: https://github.com/infinum/devproc-handbook/pull/9#discussion_r1562873145